Access grant queries (#1195)
inrupt/solid-client-access-grants-js: CHANGELOG.md

The following changes are pending, and will be applied on the next major release:

- The `getAccessGrantAll` function is deprecated. It can be replaced with `query`.

>   Although the two functions behavior is different, they can be used to achieve

Reviewer: ```suggestion
  Although the two functions' behavior is different, they can be used to achieve
```

  similar results.
- The `gConsent` and all of `gConsent/*` submodules are deprecated. The former can

>   be replaced by a regular import of the library, and the latters can be replaced

Reviewer: ```suggestion
  be replaced by a regular import of the library, and the latter can be replaced
```
Weird English but latter is also plural

  with the equivalent non-gConsent submodules (e.g. `gConsent/manage` can be replaced
  with `manage`). There is no functionality change between the two.

## Unreleased

### New feature

- Add support for custom fields. Applications are now able to read and write custom fields
  into Access Credentials (both Access Requests and Access Grants). This feature is available
  via a new option introduced in `issueAccessRequest` and `approveAccessRequest` to write the
  custom fields, and via a set of dedicated getters in the `getters/` module. A generic getter
  is introduced, `getCustomFields`, as well as a set of typed helpers, such as `getCustomInteger`.
  Typed helpers are available for integers, floats, strings and booleans.

> - Support new query endpoint: the new `query` function allows to query for Access Credentials using
>   the newly introduce ESS endpoint.

Reviewer: ```suggestion
- Support new query endpoint: the new `query` function enables querying for Access Credentials using
  the newly introduced ESS endpoint.
```
